### PR TITLE
Add pyhf proposal

### PR DESCRIPTION
**Pull request by matthewfeickert**
Add a Google Summer of Code 2020 proposal for [pyhf](https://github.com/scikit-hep/pyhf) under the [IRIS-HEP project](https://hepsoftwarefoundation.org/gsoc/projects/2020/project_IRIS-HEP.html).

This proposal PR follows (or at least tries to) the ["How to Add a New project in HSF GSoC" instructions](https://hepsoftwarefoundation.org/gsoc/guideline.html).

Noting the [HSF GSoC Mentor Guideline 2020](https://docs.google.com/document/d/110NS7iRonBUKa05pny-YaOS8mXpBYjs6DG302SeWbOI/edit?usp=sharing)'s annotated timeline:

1. Prior to the beginning of the application period:
   - [x] Think of a good project and a second (backup) co-mentor
      - @matthewfeickert (mentor)
      - @lukasheinrich (co-mentor)
      - @kratsg (co-mentor)
   - [x] Contact organization administrators about your project
      - [x] Add your project to the GSoC 2020 ideas [webpage](http://hepsoftwarefoundation.org/activities/gsoc.html)
      - [x] **Internal project submission deadline: Tuesday Feb. 4, 2020**
   - [x] Design a good test for prospective candidates
      - Demonstrate code development ability
      - Ideally a stepping stone to the project

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Note, this draft PR is being opened now as a reminder for me to get it taken care of in the morning.

**Comment by xvallspl** [comment]
Understood. Please, let us know when ready

**Comment by matthewfeickert** [comment]
> Understood. Please, let us know when ready

Will do! I'll request a review from both you and agheata.

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+## Task ideas
+
+   * Add a hardware acceleration benchmarking suite for the GPU enabled backends (TensorFlow, PyTorch, and JAX) and make comparison plots against each other and the CPU backends (`pyhf` Issues [301](https://github.com/scikit-hep/pyhf/issues/301), [348](https://github.com/scikit-hep/pyhf/issues/348)).
+   * Thoroughly document the performance benchmarking in the form of a case study.
+   * Evaluate the hardware acceleration test suite using Google TPUs.
+   * Profile the `pyhf` codebase to determine where the hardware acceleration is making the largest difference.
+
+## Expected results
+
+By the end of Summer 2020, we would like to have a robust benchmarking system for evaluating the performance of hardware accelerated `pyhf` on GPUs and TPUs, along with detailed documentation and report for the `pyhf` website.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: What means "robust benchmarking system"? Which is the minimal set of tests to be run. What does the student actually need to develop?

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Evaluation tasks
+
+Contact the mentors for a task that can be used to evaluate candidates.
+Contributing a "good first issue" PR to `pyhf` would be a rather advanced form of evaluation, but would also be an excellent test of how the student would work with the core developer team.
+
+## Requirements
+
+   * Ability to code in Python.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: Which level of experience? "Ability" does not state it, all students in GSoC are able.

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   * Good software engineering skills.
+   * Ability to work as part of a distributed team on GitHub.
+   * Familiarity with testing and documentation generation.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: Same apply to these, be more precise in terms of  tools/framework/components they need to be familiar with

**Comment by agheata** [comment]
Please also rebase on current master and fix the conflicts in mentors.md

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+layout: gsoc_proposal
+project: IRIS-HEP
+year: 2020
+organization: University of Illinois at Urbana-Champaign
+logo: pyhf-logo.png
+---
+
+## Description
+
+The [HistFactory p.d.f. template](https://cds.cern.ch/record/1456844) is per-se independent of its implementation in ROOT and it is useful to be able to run statistical analysis outside of the ROOT, RooFit, RooStats framework.
```
[the remark this answers is not in the document]
Comment (kratsg) [reply _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: resolved

**Comment by kratsg** [comment]
> Please also rebase on current master and fix the conflicts in mentors.md

conflicts fixed

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md (hunk 3 above)
Comment (kratsg) [reply _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: resolved

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md (hunk 2 above)
Comment (kratsg) [reply _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: resolved

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+The [`pyhf`](https://github.com/scikit-hep/pyhf) library is a pure-python implementation of that statistical model for multi-bin histogram-based analysis and its interval estimation is based on the asymptotic formulas of "[Asymptotic formulae for likelihood-based tests of new physics](https://arxiv.org/abs/1007.1727)".
+`pyhf` supports modern computational graph libraries such as TensorFlow and PyTorch in order to make use of features such as autodifferentiation and GPU acceleration.
+
+The goal of the project is to test and benchmark the performance increase of the hardware accelerated backends on GPUs and TPUs on openly published physics data from LHC experiments [[10.17182/hepdata.89408](https://doi.org/10.17182/hepdata.89408), [10.17182/hepdata.92006](https://doi.org/10.17182/hepdata.92006)].
```
[the remark this answers is not in the document]
Comment (matthewfeickert) [reply _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: This should be addressed now.

File: _gsocproposals/2020/proposal_IRIS-HEPpyhf.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+## Task ideas
+
+   * Add a hardware acceleration benchmarking suite for the GPU enabled backends (TensorFlow, PyTorch, and JAX) and make comparison plots against each other and the CPU backends (`pyhf` Issues [301](https://github.com/scikit-hep/pyhf/issues/301), [348](https://github.com/scikit-hep/pyhf/issues/348)).
```
[the remark this answers is not in the document]
Comment (matthewfeickert) [reply _gsocproposals/2020/proposal_IRIS-HEPpyhf.md]: This should be addressed now.

**Pull request merged**